### PR TITLE
feat: logic for conditionally rendering feedback UI

### PR DESCRIPTION
**Pull request by Mike-Heneghan**
## What

- Based on [Figma prototype](https://www.figma.com/proto/bnUUrsVRG6qPwDkTmVKACI/PlanX-Design?page-id=12766%3A9275&type=design&node-id=13794-32506&viewport=-1569%2C471%2C0.7&t=hHNmEtfB7iz7Lk6T-1&scaling=min-zoom&starting-point-node-id=13794%3A32506) conditionally render the new UI components
- Add dummy form submission methods and basic inout validation

## Why

- Sets up the paths for users to provide feedback
- Discrete chunk of work which can be tested / reviewed 

## Testing

- Testing link: https://2678.planx.pizza/testing/feedback-ui/unpublished
- As default no change to feedback banner
- As default no chance to the more info UI
- Toggle the feature `window.featureFlags.toggle("SHOW_INTERNAL_FEEDBACK")`
- Test the feedback banner behaviour works as per [Figma Prototype ](https://www.figma.com/proto/bnUUrsVRG6qPwDkTmVKACI/PlanX-Design?page-id=12766%3A9275&type=design&node-id=13794-32506&viewport=-1569%2C471%2C0.7&t=hHNmEtfB7iz7Lk6T-1&scaling=min-zoom&starting-point-node-id=13794%3A32506)
- Test the more info section works as per [Figma Prototype ](https://www.figma.com/proto/bnUUrsVRG6qPwDkTmVKACI/PlanX-Design?page-id=12766%3A9275&type=design&node-id=13794-32506&viewport=-1569%2C471%2C0.7&t=hHNmEtfB7iz7Lk6T-1&scaling=min-zoom&starting-point-node-id=13794%3A32506)

## Screen recording

https://github.com/theopensystemslab/planx-new/assets/36415632/333ffe93-e67c-4972-8e12-1403ee30e02d

### CODE REVIEW

File: editor.planx.uk/src/components/Feedback.tsx
Comment (Mike-Heneghan) [inline editor.planx.uk/src/components/Feedback.tsx:93]: I tried implementing the handler as a single function with a switch statement which I think has worked pretty well?

File: editor.planx.uk/src/components/Feedback.tsx
Comment (Mike-Heneghan) [inline editor.planx.uk/src/components/Feedback.tsx:1]: I feel like there's an opportunity to refactor this file, potentially splitting into more components and reusing more content? Although I don't want to get too bogged down in optimisations I suppose and can always come back to it later?

**Comment by Mike-Heneghan** [comment]
[Design & implement in-house feedback feature & retire Feedback Fish](https://trello.com/c/8pmX6wmZ/2545-design-implement-in-house-feedback-feature-retire-feedback-fish)

**Comment by ianjon3s** [comment]
I've had a click through test of this and it's all working perfectly from a user-perspective!

**Comment by Mike-Heneghan** [comment]
> I've had a click through test of this and it's all working perfectly from a user-perspective!

Thanks for checking this Ian! It was really helpful having your prototype for this work 🙌

File: editor.planx.uk/src/components/Feedback.tsx
Comment (DafyddLlyr) [inline editor.planx.uk/src/components/Feedback.tsx:406]: In TypeScript switch statements don't read types as nicely as they should unfortunately. 

I'd suggest adding an exhaustive check here as it would throw a type error if we added a new `FeedbackView` but didn't update this case statement.

Comment (DafyddLlyr) [reply editor.planx.uk/src/components/Feedback.tsx:406]: Tbh maybe we should enforce this with something like https://typescript-eslint.io/rules/switch-exhaustiveness-check/ to avoid having to repeatedly use an exhausive guard. Maybe a nicer solution!

File: editor.planx.uk/src/components/Feedback.tsx
Comment (DafyddLlyr) [reply editor.planx.uk/src/components/Feedback.tsx:93]: Agreed - nice and need and maintainable! Take a look at my comment below on how we could make this a little safer.

File: editor.planx.uk/src/components/MoreInfoFeedback.tsx
Comment (DafyddLlyr) [inline editor.planx.uk/src/components/MoreInfoFeedback.tsx:61]: I know this is a UI PR but just as a heads up for when you come to wire this up in a future PR - 
 - We generally use [Formik](https://formik.org/) to build and handle forms
 - We use [Yup](https://github.com/jquense/yup) when we need a schema to validation (I'd love to switch to [Zod](https://zod.dev/) but Yup and Formik play together nicely out of the box and this project pre-dates Zod)
 
We can do a handover / run though before you come to this if you'd like - there are plenty of good examples throughout the codebase also.

File: editor.planx.uk/src/components/Feedback.tsx
Comment (DafyddLlyr) [reply editor.planx.uk/src/components/Feedback.tsx:1]: This might be a nice to have in future I think - for now it feels ok to group them? We can always come back and split up (and maybe write some Storybook stories) and tests sometime soon?

**Review by DafyddLlyr — APPROVED** [review]
Super clean and nice code - a pleasure to review!

A few minor comments and suggestions, nothing blocking.

Some small UI / design thoughts (these are for @ianjon3s to think about also...!) - 

image

- The white on grey here feels quite hard to look at - is there an alternative or is this ok?

- Should we use the `ErrorWrapper` component over native browser validation (e.g. for required fields)? The benefits would be a more consistent UI and we also now capture analytics via the `ErrorWrapper` which could be useful to track.

File: editor.planx.uk/src/components/Feedback.tsx
Comment (DafyddLlyr) [reply editor.planx.uk/src/components/Feedback.tsx:86]: Alternatively, string unions (just as a type) can also work really well in these situations (TS enums can be a bit awkward / odd at times).

File: editor.planx.uk/src/ui/public/FeedbackOption.tsx
Comment (Mike-Heneghan) [reply editor.planx.uk/src/ui/public/FeedbackOption.tsx]: I'm not sure if I've full understood this one 🤔 

I've update the prop here to hopefully improve the syntax and been consistent with onClick: 7d9f353fd8ebfef1f1176904ad96efcf3509375c

File: editor.planx.uk/src/components/MoreInfoFeedback.tsx
Comment (Mike-Heneghan) [inline editor.planx.uk/src/components/MoreInfoFeedback.tsx:32]: I've also tried to reflect the language of the `feedback_type_enum` here so that it hopefully makes things easier when the forms are wired up.

File: editor.planx.uk/src/ui/public/FeedbackOption.tsx
Comment (DafyddLlyr) [reply editor.planx.uk/src/ui/public/FeedbackOption.tsx]: That's perfect - just what I had in mind 😄

File: editor.planx.uk/src/components/Feedback.tsx
Comment (DafyddLlyr) [reply editor.planx.uk/src/components/Feedback.tsx]: Yep this is a great pattern to follow 👌

File: editor.planx.uk/src/components/Feedback.tsx
Comment (DafyddLlyr) [reply editor.planx.uk/src/components/Feedback.tsx:363]: Thanks for taking a look at this!

Locally on your branch this isn't quite working for me (it's maybe a me issue!).

If I drop the final "thanks" case from this switch locally I can run `pnpm check` and it picks up an issue with this file - 

```sh
➜  editor.planx.uk git:(mh/setup-feedback-form-conditional-rendering) pnpm check

> editor.planx.uk@ check /Users/dafyddllyr/Code/planx-new/editor.planx.uk
> tsc --noEmit && pnpm lint

src/components/Feedback.tsx:363:24 - error TS2366: Function lacks ending return statement and return type does not include 'undefined'.

363   function Feedback(): FCReturn {
                           ~~~~~~~~

Found 1 error in src/components/Feedback.tsx:363
```


However my IDE doesn't pick up the affected line, it just gives an issue right at the top of the affected file - 
image

(The location of the `.tsconfig.json` should be `/Users/dafyddllyr/Code/planx-new/editor.planx.uk/tsconfig.json`)

Are you hitting these issues?

**Review by DafyddLlyr — APPROVED** [review]
One small question - might be an issue with my local setup but just wanted to cross-check.

File: editor.planx.uk/src/components/Feedback.tsx
Comment (Mike-Heneghan) [reply editor.planx.uk/src/components/Feedback.tsx:363]: > Thanks for taking a look at this!
> 
> Locally on your branch this isn't quite working for me (it's maybe a me issue!).
> 
> If I drop the final "thanks" case from this switch locally I can run `pnpm check` and it picks up an issue with this file -
> 
> ```shell
> ➜  editor.planx.uk git:(mh/setup-feedback-form-conditional-rendering) pnpm check
> 
> > editor.planx.uk@ check /Users/dafyddllyr/Code/planx-new/editor.planx.uk
> > tsc --noEmit && pnpm lint
> 
> src/components/Feedback.tsx:363:24 - error TS2366: Function lacks ending return statement and return type does not include 'undefined'.
> 
> 363   function Feedback(): FCReturn {
>                            ~~~~~~~~
> 
> Found 1 error in src/components/Feedback.tsx:363
> ```
> 
> However my IDE doesn't pick up the affected line, it just gives an issue right at the top of the affected file - image
> 
> (The location of the `.tsconfig.json` should be `/Users/dafyddllyr/Code/planx-new/editor.planx.uk/tsconfig.json`)
> 
> Are you hitting these issues?

This is a bit of a weird one 🤔 

Normally I use `npx lint-staged` before I make a commit and when I remove the `thanks` case I see:

```
❯ npx lint-staged
✔ Preparing lint-staged...
❯ Running tasks for staged files...
  ❯ package.json — 1 file
    ❯ src/**/*.{js,jsx,ts,tsx} — 1 file
      ✖ eslint --fix [FAILED]
    ✔ src/**/*.{js,jsx,ts,tsx,json,css,scss,md,html} — 1 file
    ❯ src/**/*.{ts,tsx} — 1 file
      ✖ scripts/tsc-lint.sh [KILLED]
↓ Skipped because of errors from tasks. [SKIPPED]
✔ Reverting to original state because of errors...
✔ Cleaning up temporary files...

✖ eslint --fix:

/Users/mike/planx-new/editor.planx.uk/src/components/Feedback.tsx
   95:40  warning  Unexpected any. Specify a different type                                             @typescript-eslint/no-explicit-any
   98:28  warning  Unexpected any. Specify a different type                                             @typescript-eslint/no-explicit-any
  345:12  warning  'ThanksForFeedback' is defined but never used. Allowed unused vars must match /^_/u  @typescript-eslint/no-unused-vars
  364:13  error    Switch is not exhaustive. Cases not matched: "thanks"                                @typescript-eslint/switch-exhaustiveness-check

✖ 4 problems (1 error, 3 warnings)


✖ scripts/tsc-lint.sh failed without output (KILLED).
```

When I try `pnpm check` I see:

```
Checking formatting...
All matched files use Prettier code style!
❯ pnpm check

> editor.planx.uk@ check /Users/mike/planx-new/editor.planx.uk
> tsc --noEmit && pnpm lint

src/components/Feedback.tsx:363:24 - error TS2366: Function lacks ending return statement and return type does not include 'undefined'.

363   function Feedback(): FCReturn {
                           ~~~~~~~~


Found 1 error in src/components/Feedback.tsx:363
```

In both cases I don't see any warning in my editor 🤔

Comment (Mike-Heneghan) [reply editor.planx.uk/src/components/Feedback.tsx:363]: If you have time maybe we could jump on a huddle at @DafyddLlyr as I'm slightly perplexed by this one. I think my setup/workflow hasn't been quite right.

Comment (DafyddLlyr) [reply editor.planx.uk/src/components/Feedback.tsx:363]: Yeah lets do that 👍 

Could you please drop these changes from the PR so that we can look at it in isolation, and don't hold up this PR unnecessarily?

Comment (Mike-Heneghan) [reply editor.planx.uk/src/components/Feedback.tsx:363]: I'm also thinking `FCReturn` might actually be correct as if a `case` is missing i.e. the `currentFeedbackView` is `"thanks"` then the component does indeed lack a return statement and isn't guaranteed to return a react component?

**Comment by Mike-Heneghan** [comment]
> Super clean and nice code - a pleasure to review!
> 
> A few minor comments and suggestions, nothing blocking.
> 
> Some small UI / design thoughts (these are for @ianjon3s to think about also...!) -
> 
> image
> 
> * The white on grey here feels quite hard to look at - is there an alternative or is this ok?
> * Should we use the `ErrorWrapper` component over native browser validation (e.g. for required fields)? The benefits would be a more consistent UI and we also now capture analytics via the `ErrorWrapper` which could be useful to track.

There are great spots @DafyddLlyr!

I'll defer to your judgement/update for the "Report an Issue" button @ianjon3s but we can update this in a later PR.

For the `ErrorWrapper` I think I'll add this when I wire up the forms following the pattern in your work with the design settiings @DafyddLlyr 👍

**Pull request merged**
